src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -230,6 +230,24 @@ def ok_to_stop(self, daemon_ids: List[str]) -> HandleCommandResult:
         logger.info(out)
         return HandleCommandResult(r.retval, out, r.stderr)
 
+    def _enough_daemons_to_stop(self, daemon_type: str, daemon_ids: List[str], service: str, low_limit: int) -> Tuple[bool, str]:
+        # Provides a warning about if it possible or not to stop <n> daemons in a service
+        names = [f'{daemon_type}.{d_id}' for d_id in daemon_ids]
+        out = f'It is presumed safe to stop {names}'
+        warn = False
+        number_of_daemons = len(self.mgr.cache.get_daemons_by_type(daemon_type))
+        if number_of_daemons <= low_limit:
+            warn = True
+            daemon_str = 'only daemon' if number_of_daemons == 1 else f'{number_of_daemons} daemons'
+            daemons_left_word = 'daemon' if number_of_daemons - low_limit == 1 else 'daemons'
+            daemons_left_str = 'no daemons' if number_of_daemons - \
+                low_limit == 0 else f'{str(number_of_daemons - low_limit)} {daemons_left_word}'
+            low_limit_daemons_str = f'{low_limit} daemon' if low_limit == 1 else f'{low_limit} daemons'
+            out = (f'WARNING: Stopping {daemon_str} in {service} service. '
+                   f'Service will not be operational with {daemons_left_str}. '
+                   f'At least {low_limit_daemons_str} must be running to guarantee service. ')
+        return (warn, out)
```

Comment:
had a hard time understanding this block. Maybe use something like 

```python
    def _enough_daemons_to_stop(self, daemon_type: str, daemon_ids: List[str], service: str, low_limit: int) -> Tuple[bool, str]:
        # Provides a warning about if it possible or not to stop <n> daemons in a service
        names = [f'{daemon_type}.{d_id}' for d_id in daemon_ids]
        number_of_daemons = len(self.mgr.cache.get_daemons_by_type(daemon_type))
        if number_of_daemons > low_limit:
            return False, f'It is presumed safe to stop {names}'
        
        num_daemons_left = number_of_daemons - low_limit
        
        def plural(word, plural_char, count):
            return word if count == 1 else f'{word}{plural_char}'
        
        daemon_count = "only" if number_of_daemons == 1 else number_of_daemons
        left_count = "no" if num_daemons_left == 0 else num_daemons_left

        out = (f'WARNING: Stopping {daemon_count} {plural("daemon", "s", number_of_daemons)} in {service} service.'
                f'Service will not be operational with {left_count} {plural("daemon", "s", low_limit)}. '
                f'At least {low_limit} {plural("daemon", "s", low_limit)} must be running to guarantee service. ')
        return (True, out)
``` 
instead??